x/gamm/keeper/multihop_test.go

```diff
@@ -55,32 +65,55 @@ func (suite *KeeperTestSuite) TestBalancerPoolSimpleMultihopSwapExactAmountIn()
 						TokenOutDenom: "baz",
 					},
 				},
+				poolFee:           []sdk.Dec{poolDefaultSwapFee, poolDefaultSwapFee},
 				tokenIn:           sdk.NewCoin("foo", sdk.NewInt(100000)),
 				tokenOutMinAmount: sdk.NewInt(1),
 			},
+			coinA:             sdk.NewCoin("foo", sdk.NewInt(1000000000)),
+			coinB:             sdk.NewCoin("uosmo", sdk.NewInt(1000000000)),
+			coinC:             sdk.NewCoin("baz", sdk.NewInt(1000000000)),
+			reducedFeeApplied: true,
+			expectPass:        true,
+		},
+		{
+			name: "Swap - [foo -> uosmo](pool 1) - [uosmo -> baz](pool 2) with a half fee applied, (pool 1) 1 percent fee, (pool 2) 10 percent fee",
+			param: param{
+				routes: []types.SwapAmountInRoute{
+					{
+						PoolId:        1,
+						TokenOutDenom: "uosmo",
+					},
+					{
+						PoolId:        2,
+						TokenOutDenom: "baz",
+					},
+				},
+				poolFee:           []sdk.Dec{poolDefaultSwapFee, sdk.NewDecWithPrec(1, 1)},
+				tokenIn:           sdk.NewCoin("foo", sdk.NewInt(100000)),
+				tokenOutMinAmount: sdk.NewInt(1),
+			},
+			coinA:             sdk.NewCoin("foo", sdk.NewInt(1000000000)),
+			coinB:             sdk.NewCoin("uosmo", sdk.NewInt(1000000000)),
+			coinC:             sdk.NewCoin("baz", sdk.NewInt(1000000000)),
 			reducedFeeApplied: true,
 			expectPass:        true,
 		},
 	}
 
 	for _, test := range tests {
-		// Init suite for each test.
 		suite.SetupTest()
 
 		suite.Run(test.name, func() {
 			keeper := suite.App.GAMMKeeper
-			poolDefaultSwapFee := sdk.NewDecWithPrec(1, 2) // 1%
-			poolZeroSwapFee := sdk.ZeroDec()
 
-			// Prepare pools
-			suite.PrepareBalancerPoolWithPoolParams(balancer.PoolParams{
-				SwapFee: poolDefaultSwapFee,
-				ExitFee: sdk.NewDec(0),
-			})
-			suite.PrepareBalancerPoolWithPoolParams(balancer.PoolParams{
-				SwapFee: poolDefaultSwapFee,
-				ExitFee: sdk.NewDec(0),
-			})
```

Comment:
Note to reviewer, these methods created four asset pools, making the test unnecessarily difficult to test osmo swap fee reduction. Switched to two asset pools for simplicity